stl/inc/ranges

```diff
@@ -6821,6 +6821,682 @@ namespace ranges {
+    zip_view(_RangeTypes&&...) -> zip_view<views::all_t<_RangeTypes>...>;
+
+    template <input_range... _RangeTypes>
+    inline constexpr bool enable_borrowed_range<zip_view<_RangeTypes...>> = (enable_borrowed_range<_RangeTypes> && ...);
+
+    namespace _Zip {
+        template <class _ResultType, class _LHSTuple, class _RHSTuple>
+        struct _Misc_iterator_sentinel_noexcept_solver {};
+
+        template <class _ResultType, class... _LHSTupleTypes, class... _RHSTupleTypes>
+        struct _Misc_iterator_sentinel_noexcept_solver<_ResultType, tuple<_LHSTupleTypes...>,
+            tuple<_RHSTupleTypes...>> {
+            static constexpr bool _Is_get_smallest_distance_noexcept =
+                (noexcept(_ResultType{_STD declval<_LHSTupleTypes>() - _STD declval<_RHSTupleTypes>()}) && ...)
+                && (noexcept(
+                        _ResultType{_RANGES distance(_STD declval<_LHSTupleTypes>(), _STD declval<_RHSTupleTypes>())})
+                    && ...)&& noexcept(_STD declval<_ResultType>() < _STD declval<_ResultType>())
+                && is_nothrow_copy_assignable_v<_ResultType> && is_nothrow_move_constructible_v<_ResultType>;
+        };
+
+        template <class... _LHSTupleTypes, class... _RHSTupleTypes>
+        struct _Misc_iterator_sentinel_noexcept_solver<bool, tuple<_LHSTupleTypes...>, tuple<_RHSTupleTypes...>> {
+            static constexpr bool _Is_evaluate_equality_noexcept =
+                (noexcept(_STD declval<_LHSTupleTypes>() == _STD declval<_RHSTupleTypes>()) && ...);
+        };
+
+        template <class _ResultType, class _LHSItrOrSentTuple, class _RHSItrOrSentTuple>
+            requires(tuple_size_v<decay_t<_LHSItrOrSentTuple>> == tuple_size_v<decay_t<_RHSItrOrSentTuple>>)
+        _NODISCARD constexpr _ResultType _Get_smallest_iterator_sentinel_distance(_LHSItrOrSentTuple&& _Lhs_tuple,
+            _RHSItrOrSentTuple&& _Rhs_tuple) noexcept(_Misc_iterator_sentinel_noexcept_solver<_ResultType,
+            decay_t<_LHSItrOrSentTuple>, decay_t<_RHSItrOrSentTuple>>::_Is_get_smallest_distance_noexcept);
+
+        template <class _LHSItrOrSentTuple, class _RHSItrOrSentTuple>
+            requires(tuple_size_v<decay_t<_LHSItrOrSentTuple>> == tuple_size_v<decay_t<_RHSItrOrSentTuple>>)
+        _NODISCARD constexpr bool _Evaluate_iterator_sentinel_equality(_LHSItrOrSentTuple&& _Lhs_tuple,
+            _RHSItrOrSentTuple&& _Rhs_tuple) noexcept(_Misc_iterator_sentinel_noexcept_solver<bool,
+            decay_t<_LHSItrOrSentTuple>, decay_t<_RHSItrOrSentTuple>>::_Is_evaluate_equality_noexcept);
+    } // namespace _Zip
+
+    template <input_range... _ViewTypes>
+        requires(view<_ViewTypes>&&...)
+    &&(sizeof...(_ViewTypes) > 0) template <bool _IsConst>
+    class zip_view<_ViewTypes...>::_Iterator : public _Iterator_category_container<_Iterator<_IsConst>> {
+    public:
+        // NOTE: iterator_category is conditionally present based on whether or not it exists as

```

Comment:
I initially thought that writing that there might be better because the `_Iterator` type was defined outside of the definition of `zip_view` (and thus far away from the definition of `_Iterator_category_container`), but after moving the implementation directly into the definition of the class as suggested in a different change, it becomes even more irrelevant.

Regardless, it is pretty unnecessary in retrospect.